lapce-core/src/language.rs

```diff
@@ -425,3 +410,77 @@ fn walk_tree(
         cursor.goto_parent();
     }
 }
+
+// NOTE: These tests exist only when `cargo test` is given certain `--features`
+// values, together with `-p lapce-core`. For example:
+//
+//   cargo test -p lapce-core --features lang-rust,lang-python
+//
+// will not run only `test_cpp_lang`. Or use `--all-features`:
+//
+//   cargo test -p lapce-core --all-features
+//
+// In VS Code, clicking the "Run Test" button added to `mod test` will run the
+// test functions only if RA has been given the required features (i.e. in the
+// workspace settings in .vscode/settings.json).
+//
+// If clicking the "Run Test" buttons attached to the functions, RA will add
+// `--feature lang-rust`, for example, for you to the cargo test command line,
+// in addition to any features in its workspace settings.
+#[cfg(test)]
+mod test {
+    #[test]
+    #[cfg(feature = "lang-rust")]
+    fn test_rust_lang() {
+        use super::LapceLanguage;
+        use std::path::PathBuf;
+        let path = PathBuf::from("test.rs");
+        let lang = LapceLanguage::from_path(&path);
+
+        assert!(lang.is_some());
+        let lang = lang.unwrap();
+        assert_eq!(lang, LapceLanguage::Rust);
+        let props = lang.properties();
+        assert_eq!(lang.comment_token(), props.comment);
+        assert_eq!(lang.indent_unit(), props.indent);
```

Comment:
You can extract all these checks into a helper function and call that function in all three of these tests.